var/spack/repos/builtin/packages/lapackpp/package.py

```diff
@@ -45,16 +45,27 @@ class Lapackpp(CMakePackage):
 
     def cmake_args(self):
         spec = self.spec
-        return [
+
+        args = [
             '-DBUILD_SHARED_LIBS=%s' % ('+shared' in spec),
             '-Dbuild_tests=%s'       % self.run_tests,
             '-DLAPACK_LIBRARIES=%s'  % spec['lapack'].libs.joined(';')
         ]
 
+        if spec['blas'].name == 'cray-libsci':
+            args.append(self.define('BLA_VENDOR', 'CRAY'))
+
+        return args
+
     def check(self):
         # If the tester fails to build, ensure that the check() fails.
         if os.path.isfile(join_path(self.build_directory, 'test', 'tester')):
             with working_dir(self.build_directory):
                 make('check')
         else:
             raise Exception('The tester was not built!')
+
+    def flag_handler(self, name, flags):
+        if self.spec.satisfies('%cce') and name == 'cxxflags':
+            flags.append('-DLAPACK_FORTRAN_ADD_')
```

Comment:
Isn't this flag also needed for other compilers? (e.g. gcc ?) It built fine with `cce` for me but failed to compile with `gcc` throwing an error indicating that the flag has to be passed.

Reply:
@teonnik , You're probably right, I changed `self.spec.satisfies('%cce')` to `spec['blas'].name == 'cray-libsci'`, does that fix your issue?